src/ConstructiveSolidGeometry/SurfacePrimitives/Rectangle.jl

```diff
@@ -0,0 +1,104 @@
+struct Rectangle{T,TN,TL,TW} <: AbstractSurfacePrimitive{T}
+    normal::TN
+    l::TL
+    w::TW
+    loc::T
+    function Rectangle( ::Type{T},
+                        normal::Union{Val{:x}, Val{:y}, Val{:z}},
+                        l::Union{T, <:AbstractInterval{T}},
+                        w::Union{T, <:AbstractInterval{T}},
+                        loc::T) where {T}
+        new{T,typeof(normal),typeof(l),typeof(w)}(normal,l,w,loc)
+    end
+end
+
+#Constructors
+Rectangle(b::Box{T}, normal::Val{:x}, x::Real) where {T} = Rectangle(T, normal, b.y, b.z, T(x))
+Rectangle(b::Box{T}, normal::Val{:y}, y::Real) where {T} = Rectangle(T, normal, b.x, b.z, T(y))
+Rectangle(b::Box{T}, normal::Val{:z}, z::Real) where {T} = Rectangle(T, normal, b.x, b.y, T(z))
+
+function Rectangle(;normal = Val(:z), lMin = -1, lMax = 1, wMin = -1, wMax = 1, loc = 0)
+    T = float(promote_type(typeof.((lMin, lMax, wMin, wMax, loc))...))
+    l = lMax == -lMin ? T(lMax) : T(lMin)..T(lMax)
+    w = wMax == -wMin ? T(wMax) : T(wMin)..T(wMax)
+    Rectangle(T, normal, l, w, T(loc))
+end
+
+Rectangle(normal, lMin, lMax, wMin, wMax, loc) = Rectangle(; normal = normal, lMin = lMin, lMax = lMax, wMin = wMin, wMax = wMax, loc = loc)
+
+function Rectangle(normal::Union{Val{:x}, Val{:y}, Val{:z}}, l::L, w::W, loc::C) where {L<:Real, W<:Real, C<:Real}
+    T = float(promote_type(L,W,C))
+    Rectangle(T, normal, T(l/2), T(w/2), T(loc))
+end
+
+@inline in(p::AbstractCoordinatePoint, r::Rectangle{<:Any, Val{:x}}) = begin
+    _in_y(p, r.l) && _in_z(p, r.w) && _isapprox_x(p, r.loc)
+end
+
+@inline in(p::AbstractCoordinatePoint, r::Rectangle{<:Any, Val{:y}}) = begin
+    _in_x(p, r.l) && _in_z(p, r.w) && _isapprox_y(p, r.loc)
+end
+
+@inline in(p::AbstractCoordinatePoint, r::Rectangle{<:Any, Val{:z}}) = begin
+    _in_x(p, r.l) && _in_y(p, r.w) && _eq_z(p, r.loc)
+end    
```

Comment:
By introducing aliases, this could then be 
`@inline in(p::AbstractCoordinatePoint, r::RectangleZ)`